File: frontend/src/api/fhir.ts
// FHIR API client for SEWA frontend
export async function searchPatients(name?: string, gender?: string) {
  const params = new URLSearchParams();
  if (name) params.append('name', name);
  if (gender) params.append('gender', gender);
  const res = await fetch(`/api/fhir/Patient?${params.toString()}`);
  if (!res.ok) throw new Error('Failed to fetch patients');
  return res.json();
}

export async function getPatient(patientId: string) {
  const res = await fetch(`/api/fhir/Patient/${patientId}`);
  if (!res.ok) throw new Error('Failed to fetch patient');
  return res.json();
}

File: ai-services/src/domains/fhir.py

import os
import httpx
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter(prefix="/fhir", tags=["FHIR"])

FHIR_BASE_URL = os.getenv("FHIR_BASE_URL")  # Set this env var for real API
USE_REMOTE_FHIR = os.getenv("USE_REMOTE_FHIR", "false").lower() == "true"

class FHIRPatient(BaseModel):
    id: str
    name: str
    birthDate: str
    gender: str

@router.get("/Patient/{patient_id}", response_model=FHIRPatient)
def get_patient(patient_id: str):
    if USE_REMOTE_FHIR and FHIR_BASE_URL:
        response = httpx.get(f"{FHIR_BASE_URL}/Patient/{patient_id}")
        response.raise_for_status()
        data = response.json()
        return FHIRPatient(
            id=data["id"],
            name=data["name"][0]["text"] if data.get("name") else "",
            birthDate=data.get("birthDate", ""),
            gender=data.get("gender", "")
        )
    # Local placeholder
    return FHIRPatient(id=patient_id, name="John Doe", birthDate="1980-01-01", gender="male")
